src/binaries/query/main.rs

```diff
@@ -251,3 +257,14 @@ fn run_cmd(conf: &Config) -> bool {
 
     true
 }
+
+#[cfg(not(target_os = "macos"))]
+fn try_set_max_open_files() {
```

Comment:
As a naming convention in rust, functions named like `try_xxx` usually return `Result<yyy>` to indicate that this function could return an error.

Since we just get and print a warning message here, how about renaming this function to `check_max_open_files()`?